kvserver: change replica range lease to use NodeVitality

andrewbaptist opened the pull request:
This moves all the parts of the check into the NodeVitality interface to
simplify the conditions.

Epic: none

Release note: None

Commit:
kvserver: change replica range lease to use NodeVitality

This moves all the parts of the check into the NodeVitality interface to
simplify the conditions.

Epic: none

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/105091)
<!-- Reviewable:end -->

Commit:
liveness: remove unused GetIsLiveMap

All the code has been changed to no longer require the GetIsLiveMap
function anymore.

Epic: none

Release note: None

Commit:
kvserver: remove use of IsLiveMap

As part of the effort to remove the IsLiveMap from the Store, all
non-lease related uses can be switched to directly use NodeVitality
directly.

Epic: none

Release note: None

Commit:
kvserver: create node liveness in tests

Previously node liveness wasn't constructed in tests. This commit adds
it to allow removing a number of nil checks.

Epic: none

Release note: None
